kratos/geometries/coupling_geometry.h

```diff
@@ -337,6 +304,236 @@ template<class TPointType> class CouplingGeometry
+                mpGeometries[1]->ProjectionPoint(
+                    integration_points_global_coords_vector[j],
+                    global_slave_coords,
+                    local_slave_coords);
+
+                integration_points_slave[j][0] = local_slave_coords[0];
+                integration_points_slave[j][1] = local_slave_coords[1];
+                integration_points_slave[j][2] = local_slave_coords[2];
+            }
+        }
+
+        GeometriesArrayType slave_quadrature_points(num_integration_points);
+        mpGeometries[1]->CreateQuadraturePointGeometries(
+            slave_quadrature_points,
+            NumberOfShapeFunctionDerivatives,
+            integration_points_slave,
+            rIntegrationInfo);
+
+        for (SizeType i = 0; i < num_integration_points; ++i) {
+            KRATOS_DEBUG_ERROR_IF(norm_2(master_quadrature_points(i)->Center() - slave_quadrature_points(i)->Center()) > model_tolerance)
+                << "Difference between master and slave coordinates above model tolerance of " << model_tolerance
+                << ". Location of master: " << master_quadrature_points(i)->Center() << ", location of slave: "
+                << slave_quadrature_points(i)->Center() << ". Distance: "
+                << norm_2(master_quadrature_points(i)->Center() - slave_quadrature_points(i)->Center()) << std::endl;
+
+            rResultGeometries(i) = Kratos::make_shared<CouplingGeometry<PointType>>(
+                master_quadrature_points(i), slave_quadrature_points(i));
+        }
+
+        KRATOS_ERROR_IF(mpGeometries.size() > 2)
+            << "CreateQuadraturePointGeometries not implemented for coupling of more than 2 geomtries. "
+            << mpGeometries.size() << " are given." << std::endl;
+    }
+
+    ///@}
+    ///@name Span Utilities
+    ///@{
+
+    static void MergeSpans(
+        std::vector<double>& rResultSpans,
+        const std::vector<double>& rSpans1,
+        const std::vector<double>& rSpans2,
+        double Tolerance = 1e-6)
+    {
+        NurbsInterval interval_1(rSpans1[0], rSpans1[rSpans1.size() - 1]);
+        NurbsInterval interval_2(rSpans2[0], rSpans2[rSpans2.size() - 1]);
+
+        for (IndexType i = 0; i < rSpans1.size(); ++i) {
+            double temp = rSpans1[i];
+            interval_2.IsInside(temp);
+            rResultSpans.push_back(temp);
+        }
+        for (IndexType i = 0; i < rSpans2.size(); ++i) {
+            double temp = rSpans2[i];
+            interval_1.IsInside(temp);
+            rResultSpans.push_back(temp);
+        }
+
+        SortUnique(rResultSpans, Tolerance);
+    }
+
+    static void SortUnique(
+        std::vector<double>& rResultSpans,
+        const double Tolerance)
+    {
+        std::sort(std::begin(rResultSpans), std::end(rResultSpans));
+
+        auto last = std::unique(std::begin(rResultSpans), std::end(rResultSpans),
+            [=](double a, double b) { return b - a < Tolerance; });
```

Comment:
Aren't you capturing here the vector as well (by value)? Not only the tolerance?

Reply:
Not sure what you mean. It works like this ;)

Reply:
I mean, it should be enough if you only capture the Tolerance variable in [].